### PR TITLE
feat(035): API threshold — keep light keys on metered API

### PR DESCRIPTION
**Pull request by studert**
## What & why

The API→Subscription calculator's **right-sized** scenario used to force *every* key onto a flat seat — even a key burning $3/mo got a $25 Standard seat, which costs **more** than leaving it on the API. This adds a lower **API threshold**: below it, a key stays on pay-as-you-go metered API.

It mirrors the existing **Premium threshold** and **defaults to $25** — the Standard seat price, i.e. the break-even below which a seat can never pay off. The result is a genuine three-band right-sizing (**API · Standard · Premium**) and a right-sized total that is now provably ≤ the metered baseline at default thresholds.

## Changes

- **Engine** (`src/lib/scenarios/api-subscription.ts`): `SeatTier` gains `"api"`; `ScenarioInputs.apiThresholdCents`; `ScenarioResult.apiCount`. `mapSeat` is Premium-first; an `"api"` key carries its own spend as `seatCents` (zero delta, still foots the right-sized total). `standardCount = count − premiumCount − apiCount`.
- **Client** (`api-subscription-client.tsx`): a second slider paired with the Premium one (clamped so the floor can't exceed the ceiling); three-way readouts across the threshold cell, KPI, verdict (which omits the "keeping N keys" clause when none are metered), scenario card, comparison bar, and table footer; a new `API` `SeatPill` variant. `/simplify` pass extracted a `ThresholdSlider` component and `SEAT_PILL` / `TIER_SORT_ORDER` lookups.
- **Tests**: boundary cases + three-band anchors, plus an `apiThreshold = 0` superset test pinning the legacy figures.

**No schema change** — read-only over existing tables. No new dependency.

## Regression anchors (recomputed on the 2026-06-09 fixture)

| Population | API · Std · Prem | Right-sized /mo | vs baseline |
|---|---|---|---|
| All 47 | 16 · 22 · 9 | **$1,775.91** (was $2,075) | **−42%** (was 32%) |
| Active 43 | 16 · 20 · 7 | **$1,475.91** | −39% |
| `apiThreshold=0` (superset) | 0 · 38 · 9 | $2,075 | legacy, pinned in tests |

Note: the default headline figures **intentionally change** because the floor defaults to $25 — the two shipped anchors were updated and a superset test preserves the old numbers.

## Verification

- `pnpm test` (22 passed) · `pnpm typecheck` clean · `pnpm lint` clean (`--max-warnings 0`).
- 4-way adversarial review (engine math / test coverage / UI completeness / quality+a11y+tokens) → all pass.
- **In-browser** (Playwright, agent session): default renders **16 API · 22 Standard · 9 Premium → $1,776/mo, 42% cut**; dragging the floor to $0 reverts to legacy 38-Standard; clamp confirmed (Premium → $20 drags the floor down); 16 `API` pills in the table.

## Docs & open questions

Plan + running notes under `specs/035-scenario-calculators/` (`api-threshold-implementation-plan.html`, `implementation-notes.html`). Open questions for review: the **"Keep on API <"** label wording; whether to refresh the standalone `prototype.html` to the three-band split.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/lib/scenarios/api-subscription.ts [hunk 1]
```diff
@@ -79,17 +82,24 @@ export function usageForUser(
...
 }
 
-/** Map a monthly spend onto the seat tier it qualifies for. */
+/**
+ * Map a monthly spend onto the cheapest viable option among
+ * {metered API, Standard seat, Premium seat}. Premium-first so the bands stay
+ * sane even if the thresholds are dragged out of order; the "api" tier carries
+ * the user's own spend as its `seatCents`, so it contributes its real burn to
+ * the right-sized total and nets a zero delta against the metered baseline.
+ */
```
[the remark this answers is not in the document]
Comment (studert) [reply src/lib/scenarios/api-subscription.ts:100]: Good catch — fixed in cfcceeb. The JSDoc no longer claims "cheapest viable option"; it now spells out the three threshold rules (`usage >= premiumThreshold` → Premium; `usage < apiThreshold` → metered API; otherwise Standard, Premium-first) and adds an explicit note that these are *policy* thresholds: they coincide with the cheapest option only at the break-even defaults (API floor = Standard price, Premium threshold = Premium price), and dragged away from those a tier can be assigned even when another would cost less.

File: src/app/scenarios/api-subscription/api-subscription-client.tsx [hunk 2]
```diff
@@ -427,8 +469,14 @@ export function ApiSubscriptionClient({
...
-          Moving all {result.count} API users to {result.premiumCount} Premium +{" "}
-          {result.standardCount} Standard seats costs{" "}
+          Moving {result.count} API users to {result.premiumCount} Premium +{" "}
+          {result.standardCount} Standard seats
+          {result.apiCount > 0
+            ? ` and keeping ${result.apiCount} light ${
+                result.apiCount === 1 ? "key" : "keys"
+              } on metered API`
+            : ""}{" "}
+          costs{" "}
```
[the remark this answers is not in the document]
Comment (studert) [reply src/app/scenarios/api-subscription/api-subscription-client.tsx]: Fixed in cfcceeb. The verdict is now built as a true partition — a small `joinParts()` helper joins only the non-empty groups — so it never implies all keys move to seats. Verified in-browser across the mixes that exercise the edge cases:

- default → "Right-sizing the 47 API users — **9 Premium, 22 Standard, and 16 kept on metered API** — costs $1,776/mo, a 42% cut…"
- API floor $0 → "…— **9 Premium and 38 Standard** —…" (API group omitted)
- Standard-empty clamp (Premium = $20) → "…— **32 Premium and 15 kept on metered API** —…" (Standard group omitted)

Each breakdown now sums to the count in scope (47).

File: src/lib/scenarios/api-subscription.ts [hunk 3]
```diff
@@ -23,11 +23,13 @@ export type ScenarioInputs = {
   premiumCents: number;
   /** usage >= threshold ⇒ Premium seat, otherwise Standard. */
   premiumThresholdCents: number;
+  /** usage < threshold ⇒ keep the key on metered API (no seat migration). */
+  apiThresholdCents: number;
```
[the remark this answers is not in the document]
Comment (studert) [reply src/lib/scenarios/api-subscription.ts]: Fixed in 6a3577e. The `premiumThresholdCents` doc no longer says "otherwise Standard" — it now reads "usage >= threshold ⇒ Premium; below it the key is Standard or, under `apiThresholdCents`, the metered API tier (see mapSeat)", so the three-band behaviour is visible right at the field.

File: src/app/scenarios/api-subscription/api-subscription-client.tsx [hunk 4]
```diff
@@ -427,12 +486,12 @@ export function ApiSubscriptionClient({
           </span>
         </p>
         <p className="max-w-3xl text-sm text-muted-foreground">
-          Moving all {result.count} API users to {result.premiumCount} Premium +{" "}
-          {result.standardCount} Standard seats costs{" "}
+          Right-sizing the {result.count} API{" "}
+          {result.count === 1 ? "user" : "users"} — {seatBreakdown} — costs{" "}
           <span className="text-ink">
```
[the remark this answers is not in the document]
Comment (studert) [reply src/app/scenarios/api-subscription/api-subscription-client.tsx]: Good edge-case — fixed in 6a3577e. The breakdown is now folded into a `verdictLead` string whose interjection is conditional, so there are no dangling dashes when the population is empty:

- `count = 0` → "Right-sizing the 0 API users costs $0/mo, the same as the $0/mo API run-rate."
- `count = 1` → "Right-sizing the 1 API user — … —" (singular)
- default unchanged → "Right-sizing the 47 API users — 9 Premium, 22 Standard, and 16 kept on metered API — costs …"

File: src/lib/scenarios/api-subscription.ts [hunk 5]
```diff
@@ -79,17 +82,33 @@ export function usageForUser(
...
-    tier,
-    seatCents: tier === "premium" ? inputs.premiumCents : inputs.standardCents,
-  };
+  if (usageCents >= inputs.premiumThresholdCents) {
+    return { tier: "premium", seatCents: inputs.premiumCents };
+  }
+  if (usageCents < inputs.apiThresholdCents) {
+    return { tier: "api", seatCents: usageCents };
+  }
+  return { tier: "standard", seatCents: inputs.standardCents };
```
[the remark this answers is not in the document]
Comment (studert) [reply src/lib/scenarios/api-subscription.ts:111]: Fixed in 08c4622. The per-row "Seat $/mo" cell now uses `formatCurrency` for API-tier rows (whole-dollar seat prices keep `formatUSD0`), so it matches the cents-precise "API basis" cell exactly. Verified in-browser:

| User | API basis | Seat $/mo | Δ |
|---|---|---|---|
| (API row) | $23.82 | $23.82 | — |
| (API row) | $16.33 | $16.33 | — |

No more $24-vs-$23.82 mismatch beside a "—" delta.

File: src/app/scenarios/api-subscription/api-subscription-client.tsx [hunk 6]
```diff
@@ -56,8 +57,35 @@ function basisLabel(key: string, ds: ApiSubscriptionDataset): string {
...
+
 type SortKey = "name" | "status" | "usage" | "tier" | "seat" | "delta";
 
+// Seat-tier presentation kept in one place so the three-tier model stays
+// first-class: sort rank (cheapest → priciest) and per-tier pill styling.
+const TIER_SORT_ORDER: Record<SeatTier, number> = {
+  api: 0,
+  standard: 1,
+  premium: 2,
+};
```
[the remark this answers is not in the document]
Comment (studert) [reply src/app/scenarios/api-subscription/api-subscription-client.tsx:76]: Fixed in 08c4622. Reworded the comment to "sort rank (tier escalation API → Standard → Premium, not a cost ordering — see mapSeat)", so it no longer implies a price relationship and stays consistent with the engine doc's "these are policy thresholds, not a price comparison" note.
